### ISSUE TITLE
Ability to include documents without a \begin{sile} ... \end{sile} environment

### ISSUE THREAD

**Issue by baubleb** [sile-typesetter/sile#1413]
## Ability to include documents without a \begin{sile} ... \end{sile} environment

Having to wrap every included sub document in a *sile* environment is document ceremony to fit the requirement of having a root node to be valid XML. However, we are not coding XML documents, we are coding SILE documents. This places burden onto the document writer to conform to XML requirements.

When sile compiles the document, when it finds no sile environment wrapping on an included text, it could wrap that itself thus freeing the document writer from the XML conformance burden yet adhering to XML rigour that it requires itself.

**Comment by Omikhleia** [comment]
"Plain unmarked text" is something different, it could be a ".txt" rather than ".sil" and we could have an input reader for that (or I wonder what one expects if a plain unmarked text file contains percents, backslashes etc.)

**Comment by baubleb** [comment]
> "Plain unmarked text" is something different, it could be a ".txt" rather than ".sil"

It is quite normal to have all text files with and without markup just labelled .tex (or .sil in this case). I might want to decide to add markup to the unmarkedup text later. Whether it has sile commands/environments in the text should not force me to have to rename it from .txt to .sil, that will cause problems with version control systems, we would use SVN or Git for example.

Having to dance with file extensions is just asking for trouble.

**Comment by baubleb** [comment]
Yes, thank you, this works, a simple `\input` command solves it.

test.sil
```
\begin{document}
\script[src = ./commands]

\input{testinput.sil}

\end{document}
```

testinput.sil

```
included text

\center{text}
```

commands.lua
```
SILE.registerCommand("input", function(_, content)
    local inputFile = io.open(content[1], "r")
    io.input(inputFile)
    local inputContents = io.read("*a")
    io.close(inputFile)

    SILE.doTexlike(inputContents)
end)
```

**Comment by alerque** [comment]
Does the issue with trailing comments go away if you add a newline automatically, like this:

```lua
SILE.doTexlike(inputContents.."\n")
```

**Comment by baubleb** [comment]
> Does the issue with trailing comments go away if you add a newline automatically, like this:
> 
> ```lua
> SILE.doTexlike(inputContents.."\n")
> ```

Yes, that works.

**Comment by alerque** [comment]
This is going to be much more sensible in the next major release (v0.14.0). The `SILE.doTexLike()` function is going away, and any recognized input format can be used as an include. You can use `SILE.proccessString(content, format)` where format is optional to autodetect (SIL, XML, Lua) or can be specified, so `SILE.processString(content, "sil")` should be a one to one replacement for `SILE.doTexLike(content)`.

Additionally `\include[src=]` will read and process a file (or inline content) with an optional `[format=...]` argument to skip file type detection. For more direct access to the same function you will also have `\sil{content}` and `\xml{content}`.

The top level wrapper (*document* for SIL or *sile* for XML) is also optional for includes and only required for the master document. It can be skipped for includes and will be skipped over when processing if included in a document not at the root.

**Comment by ctrlcctrlv** [comment]
Awesome! :tada:
